### ISSUE TITLE
DNS examples fails with panic

### ISSUE THREAD

**Issue by danielpacak** [inspektor-gadget/inspektor-gadget#2452]
## Description

When I compile and run the DNS example it fails with Go panic.

## Impact

I'm using DNS tracer in a custom agent, which fails with the same error. Therefore, I cannot upgrade to the latest and greatest release.

## Environment and steps to reproduce

I believe it does not depend on environment. Just go build and run the dns executable binarary.
It's reproducible in the latest release v0.25.0:

```
cd examples/gadgets/basic/trace/dns/
go build
```

``` console
$ sudo ./dns 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x2042819]

goroutine 21 [running]:
github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/dns/tracer.startGarbageCollector.func1()
	/home/dpacak/go/src/github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/dns/tracer/gc.go:63 +0xd9
created by github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/dns/tracer.startGarbageCollector in goroutine 1
	/home/dpacak/go/src/github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/dns/tracer/gc.go:51 +0x156
```

## Expected behavior

The DNS tracer works without errors.

## Additional information

The `startGarbageCollector` method is invoked with the following args, but the `t.ctx` hasn't been set. It's only set by the `Init` method if it's invoked explicitly on the tracer instance.

``` go
startGarbageCollector(t.ctx, logger, dnsTimeout, queryMap) 
```

**Comment by flyth** [comment]
Thanks for the report! I fixed that when implementing #2443 but forgot to mention it.

**Comment by mauriciovasquezbernal** [comment]
Fixed by https://github.com/inspektor-gadget/inspektor-gadget/pull/2443.
